File: lwjgl/src/main/java/com/darkraha/opengldemoj/renders/Render.java
package com.darkraha.opengldemoj.renders;

import com.darkraha.opengldemoj.gl.AppOGL;

import static org.lwjgl.opengl.GL33.*;

/**
 * Empty renderer, can be used as base class for other renderers.
 */
public class Render {

    public static final float PI_F = (float) Math.PI;
    public static final float TO_RAD = PI_F / 180.f;
    public static final float ALNGLE45 = (float) Math.toRadians(45.0f);

    protected float surfaceWidth = 0f;
    protected float surfaceHeight = 0f;
    protected float aspect = 0f;

    protected void setSurfaceSize(int w, int h) {
        surfaceWidth = w;
        surfaceHeight = h;
        aspect = surfaceWidth / surfaceHeight;
    }


    public void onSetup(AppOGL appOGL) {
        setSurfaceSize(appOGL.getWidth(), appOGL.getHeight());
        glClearColor(0.0f, 0.0f, 0.0f, 0.0f);
        glEnable(GL_DEPTH_TEST);
    }

    public void onSurfaceChanged(AppOGL appOGL, int w, int h) {
        setSurfaceSize(w, h);
        glViewport(0, 0, w, h);
    }

    public void onDrawFrame(AppOGL appOGL) {
        glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
    }

    public void onDispose(AppOGL appOGL) {

    }

}

File: lwjgl/src/main/kotlin/com/darkraha/opengldemokt/renders/Render.kt
package com.darkraha.opengldemokt.renders

import com.darkraha.opengldemokt.gl.AppOGL
import org.lwjgl.opengl.GL33.*

open class Render {

    protected var surfaceWidth = 0f
    protected var surfaceHeight = 0f
    protected var aspect = 0f

    open fun onSetup(appOGL: AppOGL) {
        setSurfaceSize(appOGL.width, appOGL.height)
        glClearColor(0.0f, 0.0f, 0.0f, 1.0f)
        glEnable(GL_DEPTH_TEST)
    }

    protected fun setSurfaceSize(w: Int, h: Int){
        surfaceWidth = w.toFloat()
        surfaceHeight = h.toFloat()
        aspect = surfaceWidth/surfaceHeight
    }

    open fun onSurfaceChanged(appOGL: AppOGL, width: Int, height: Int) {
        setSurfaceSize(width,height)
        glViewport(0, 0, width, height)
    }

    open fun onDrawFrame(appOGL: AppOGL) {
        glClear(GL_COLOR_BUFFER_BIT or GL_DEPTH_BUFFER_BIT)
    }

    open fun onDispose(appOGL: AppOGL) {

    }

    companion object {
        @JvmStatic
        val PI_F = Math.PI.toFloat()

        @JvmStatic
        val TO_RAD = PI_F / 180.0f

        @JvmStatic
        val ALNGLE45 = 45f * TO_RAD
    }
}